src/goto-cc/compile.cpp

```diff
@@ -504,8 +504,23 @@ bool compilet::parse(
       error() << "PARSING ERROR" << eom;
       return true;
     }
+    // Question: Should this move after a typecheck. I don't think so,
+    // because typecheck should also typecheck the contracts in
+    // theory, so it is actually even better to let typecheck happen
+    // after this transformation so that we make sure that the
+    // transformed contracts are correct
```

Comment:
I think what we call "typecheck" actually isn't a type check per se, but as far as that's concerned if we only did a pure typecheck we should really do it _both_ before and after the transformation because if something is wrong after the transformation we'd like to know if was wrong to begin with or if the transformation messed it up.